utils/helpers.ts

```diff
@@ -1,33 +1,26 @@
 import { ethers } from "ethers";
+import GenericERC20TokenAbi from "../artifacts/GenericERC20Token.json";
 
-const ETH = "0x0000000000000000000000000000000000000000";
-const minABI = [
-  // balanceOf
-  {
-    constant: true,
-    inputs: [{ name: "_owner", type: "address" }],
-    name: "balanceOf",
-    outputs: [{ name: "balance", type: "uint256" }],
-    type: "function",
-  },
-];
+export const ETH = "0x0000000000000000000000000000000000000000";
 
 export const getMetamaskBalance = async (ethereum, tokenAddress, account) => {
   const provider = new ethers.providers.Web3Provider(ethereum);
   const signer = provider.getSigner();
-  let balance;
+  let balance, decimals;
 
   if (tokenAddress === ETH) {
     balance = await provider.getBalance(account);
   } else {
     const token: ethers.Contract = new ethers.Contract(
       tokenAddress,
-      minABI,
+      GenericERC20TokenAbi,
       signer
     );
 
+    decimals = await token.decimals();
+
     balance = await token.balanceOf(account);
   }
 
-  return Number(ethers.utils.formatEther(balance));
+  return Number(ethers.utils.formatUnits(balance, decimals));
```

Comment:
not sure how this format function behaves, if decimals is undefined (Eth case) will it fallback to 18?

Reply:
in ETH case we get balance from ethers provider 
`if (tokenAddress === ETH) {
    balance = await provider.getBalance(account);
  }`